11-payment-encoding.md

```diff
@@ -115,25 +115,19 @@ A writer MUST include exactly one `p` field, and set `payment_hash` to
 the SHA-2 256-bit hash of the `payment_preimage` which will be given
 in return for payment.
 
-A writer MUST NOT include more than one `d`, `h`, or `x` fields, and
-MAY include more than one `f` field.
-
-A writer MUST include either a `d` or `h` field, and MUST NOT include
-both.  If included, a writer SHOULD make `d` a complete description of
+A writer MUST include either exactly one `d` or exactly one `h` field.  If included, a 
+writer SHOULD make `d` a complete description of
 the purpose of the payment.  If included, a writer MUST make the preimage
 of the hashed description in `h` available through some unspecified means,
 which SHOULD be a complete description of the purpose of the payment.
 
-A writer SHOULD use the minimum `x` `data_length` possible.
+A writer MAY include an `x` field, which SHOULD use the minimum `data_length` 
+possible.
```

Comment:
Well I used the same terminology as for the `n` field, "MAY include an `..` field", if that is not clear we should change both.

How about changing both to "MAY include *one* `..` field", it may be less ambiguous and is consistent with the wording of other paragraphs?